internal/node/controller/pds.go

```diff
@@ -0,0 +1,130 @@
+package controller
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	types "github.com/eagraf/habitat-new/core/api"
+
+	"github.com/eagraf/habitat-new/internal/node/config"
+)
+
+type PDSClientI interface {
```

Comment:
nit, but I think the naming here could be a little better. How do you feel about
```
type Client interface {} // remove "PDS" since other packages will already do 'pds.Client'

type ClientForNode struct {}
```

or something similar.
Also why does this need to be an interface?

Reply:
The interface is for easy mocking. I have the `PDS` in front because this is still in the `controller` package. Ideally, we use the client from `indigo` once that is integrated into our repo